pinot-core/src/main/java/com/linkedin/pinot/core/query/selection/SelectionOperatorUtils.java

```diff
@@ -516,55 +516,57 @@ public static JSONArray getJSonArrayFromRow(Serializable[] poll, List<String> se
   public static Serializable[] getFormattedRow(Serializable[] inputValues, List<String> selectionColumns,
       DataSchema dataSchema)
       throws JSONException {
-    Serializable[] formattedRow = new Serializable[inputValues.length];
-
-    for (int i = 0; i < dataSchema.size(); ++i) {
-      // If column not part of schema, we don't know how to format it, so just skip.
-      if (!selectionColumns.contains(dataSchema.getColumnName(i))) {
+    Serializable[] formattedRow = new Serializable[selectionColumns.size()];
+
+    for (int i = 0; i < selectionColumns.size(); ++i) {
+      int idxInDataSchema = getColumnIdxInDataSchema(selectionColumns.get(i), dataSchema);
```

Comment:
instead of doing this for every column, compute the columnName to index map once outside the loop